server/src/ocv.rs

```diff
@@ -43,6 +43,116 @@ impl Ocv {
+      cached_votes.to_vec()
+    } else {
+      let transactions = self.archive.fetch_transactions(start_time, end_time)?;
+
+      let chain_tip = self.archive.fetch_chain_tip()?;
+      let votes = Wrapper(transactions.into_iter().map(std::convert::Into::into).collect())
+      .process_mep(&id.to_string(), chain_tip)
+      .sort_by_timestamp()
+      .to_vec()
+      .0;
+
+      self.caches.votes.insert(proposal_key.clone(), Arc::new(votes.clone())).await;
+      tracing::info!("votes {}", votes.len());
+      votes
+    };
+  
+    // weighted votes
+    let mut positive_stake_weight = Decimal::from(0);
+    let mut negative_stake_weight = Decimal::from(0);
+
+    // check community votes
+    let mut total_positive_community_votes = 0;
+    let mut total_negative_community_votes = 0;
+    for vote in &votes {
+      if vote.memo.to_lowercase() == format!("yes id {}", id.to_string()) {
+        total_positive_community_votes += 1;
+      }
+      if vote.memo.to_lowercase() == format!("no id {}", id.to_string()) {
+        total_negative_community_votes += 1;
+      }
+    }
+     // Check if enough positive votes
+     if total_positive_community_votes < 10 {
+      return Ok(GetMinaProposalConsiderationResponse {
+        proposal_id: id,
+        total_community_votes: votes.len(),
+        total_positive_community_votes,
+        total_negative_community_votes,
+        total_stake_weight: Decimal::ZERO,
+        positive_stake_weight: Decimal::ZERO,
+        negative_stake_weight: Decimal::ZERO,
+        votes,
+        elegible: false,
+        vote_status: "Insufficient voters".to_string(),
+    });
+    }
+
+    // Calculate weighted votes 
+    // Calculate weighted votes if ledger_hash params is provided
+    if let Some(hash) = ledger_hash {
+      let votes_weighted = if let Some(cached_votes) = self.caches.votes_weighted.get(&proposal_key).await {
+        cached_votes.to_vec()
+      } else {
+        let transactions = self.archive.fetch_transactions(start_time, end_time)?;
+
+        let chain_tip = self.archive.fetch_chain_tip()?;
+
+        let ledger = if let Some(cached_ledger) = self.caches.ledger.get(&hash).await {
+          Ledger(cached_ledger.to_vec())
+        } else {
+          let ledger = Ledger::fetch(self, &hash).await?;
+
+          self.caches.ledger.insert(hash, Arc::new(ledger.0.clone())).await;
+
+          ledger
+        };
+
+        let votes = Wrapper(transactions.into_iter().map(std::convert::Into::into).collect())
+        .into_weighted_mep(&id, &ledger, chain_tip)
+        .sort_by_timestamp()
+        .0;
+
+        self.caches.votes_weighted.insert(proposal_key.clone(), Arc::new(votes.clone())).await;
+
+        votes
+      };
+      for vote in &votes_weighted {
+        if vote.memo.to_lowercase() == format!("no id {}", id.to_string()) {
+              negative_stake_weight += vote.weight;
+          }
+          if vote.memo.to_lowercase() == format!("yes id {}", id.to_string()) {
```

Comment:
same as in other places, `usize` implements Display 
```suggestion
          if vote.memo.to_lowercase() == format!("yes id {}", id) {
```

Reply:
fixed